tests/integration/replication-4.tcl

```diff
@@ -85,6 +85,40 @@ start_server {tags {"repl external:skip"}} {
     }
 }
 
+start_server {tags {"repl external:skip"}} {
+    start_server {} {
+        set master [srv -1 client]
+        set master_host [srv -1 host]
+        set master_port [srv -1 port]
+        set slave [srv 0 client]
+
+        test {First server should have role slave after SLAVEOF} {
+            $slave slaveof $master_host $master_port
+            wait_for_condition 50 100 {
+                [s 0 master_link_status] eq {up}
+            } else {
+                fail "Replication not started."
+            }
+        }
+
+        test {Replication of an expired key does not delete the expired key} {
```

Comment:
https://github.com/enjoy-binbin/redis/runs/4440834343?check_suite_focus=true#step:4:5814

the test failed once on my CI (FreeBSD), looks like was a timing issue. 
I took a look, here are my thoughts: 
- the key expired before `INCR` was executed, because the execution time of `wait_for_ofs_sync` exceeded one second. (or maybe related the KILL).

I didn't think of a good way:
1. maybe we can use $slave debug sleep ? (But there are also such problems)
2. add a retry

> Check that k is locigally expired but is present in the replica.

also there was a typo (locigally)

Reply:
Good catch! Thanks for looking into it.

I don't understand how debug sleep can help...?

Add a retry, yes I guess it can work. We can double the expire time at every retry, so starting with 1 second, then 2, 4, 8 etc. We can add a check before the first `wait_for_ofs_sync` that k didn't expire and if it did, we retry. WDYT?

Reply:
oh i mean that maybe we can use debug sleep to replace the kill

as for retry. The double is a good idea. Maybe we can start from Ms, reduce the running time.